app/components/pipeline/workflow/component.js

```diff
@@ -170,6 +193,21 @@ export default class PipelineWorkflowComponent extends Component {
       );
     }
 
+    if (hasStages) {
+      const stageBuilds = this.workflowDataReload.getStageBuildsForEvent(
+        event.id
+      );
+
+      if (!stageBuilds) {
+        this.workflowDataReload.registerStageBuildsCallback(
+          STAGE_BUILD_QUEUE_NAME,
+          event.id,
+          this.stageBuildsCallback
+        );
+      }
+      this.stageBuilds = stageBuilds;
```

Comment:
```suggestion
      } else {
        this.stageBuilds = stageBuilds;
      }
```
Wrap this in an `else` to prevent any possible race condition where the `undefined` value of `stageBuilds` overwrites the actual fetched data that was set in the callback.

Reply:
Moved it before to the `if` block. When the `event` is changed, we should reset the `stageBuilds` before the callback is satisfied.